### PR TITLE
fix(scanner): prevent NaN avg dollar volume failures

### PR DESCRIPTION
**Pull request by kjpou1**
## Summary

Prevent snapshot scan failures when `avg_dollar_volume` cannot be calculated due to NaN values.

## Problem

Daily snapshot generation was quarantining with:

* 9,979 symbols processed
* 6,741 `scanner_error_result` failures
* 3,238 rows produced

Investigation traced the failures to:

```python
result["avg_dollar_volume"] = int(avg_volume * current_price)
```

where either `avg_volume` or `current_price` could contain NaN values.

This resulted in:

```text
ValueError: cannot convert float NaN to integer
```

and caused the entire symbol scan result to be marked as an error.

## Fix

Handle invalid values during `avg_dollar_volume` calculation instead of allowing the exception to propagate and fail the scan.  #257 

## Impact

Prevents valid symbol scans from being discarded due to NaN values encountered while calculating the derived `avg_dollar_volume` field.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved the robustness of average dollar volume calculations by only computing the value when both required inputs are valid finite numbers; non-finite values no longer lead to incorrect results.
  * Added debug logging to clearly indicate when average dollar volume is skipped due to invalid or missing input data, helping diagnose edge cases faster.

### CODE REVIEW

**Comment by kjpou1** [comment]
Test Bootstrap from scratch

image

operations

image

Here is the daily 


image




here is the log

l Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/149.0.0.0 Safari/537.36" "-"
celery-marketjobs-us-1  | [2026-06-22 10:21:45,286: INFO/ForkPoolWorker-1] Scan 19752c76-cf5a-45d3-a8d5-d34b3fb5c74b: querying feature_store (run_id=1)
celery-marketjobs-us-1  | [2026-06-22 10:21:45,541: INFO/ForkPoolWorker-1] Scan 19752c76-cf5a-45d3-a8d5-d34b3fb5c74b: querying filter options from feature_store (run_id=1)
celery-marketjobs-us-1  | [2026-06-22 10:21:48,096: INFO/ForkPoolWorker-1] Scan 19752c76-cf5a-45d3-a8d5-d34b3fb5c74b: querying feature_store (run_id=1)
celery-marketjobs-us-1  | [2026-06-22 10:21:48,125: INFO/ForkPoolWorker-1] Created auto scan 19752c76-cf5a-45d3-a8d5-d34b3fb5c74b for published feature run 1
celery-marketjobs-us-1  | [2026-06-22 10:21:48,128: INFO/ForkPoolWorker-1] Task app.interfaces.tasks.feature_store_tasks.build_daily_snapshot[6cf4d9aa-4d7c-4ea6-9d68-fc75c760916b] succeeded in 2302.1522306299885s: {'run_id': 1, 'status': 'published', 'as_of_date': '2026-06-18', 'market': 'US', 'total_symbols': 9979, 'processed_symbols': 9979, 'failed_symbols': 19, 'skipped_symbols': 13, 'row_count': 9960, 'duration_seconds': 2288.98, 'dq_passed': True, 'auto_scan_id': '19752c76-cf5a-45d3-a8d5-d34b3fb5c74b', 'metadata_refresh': {'run_id': 1, 'ranking_date': '2026-06-18', 'total_rows': 9960, 'updated_rows': 7197, 'missing_industry_rows': 2763, 'missing_rank_rows': 1277}, 'cleaned_stale_runs': 0, 'warnings': ['Skipped unsupported Yahoo price symbols in bootstrap cache-only snapshot: 13', 'Mean score 16.6022 outside expected range [20.0, 80.0]'], 'failure_diagnostics': {'reason_counts': {'scanner_error_result': 19}, 'samples': [{...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}], 'sample_limit': 50}}
celery-general-1        | [2026-06-22 10:21:48,128: INFO/MainProcess] Task app.tasks.runtime_bootstrap_tasks.complete_local_runtime_bootstrap[f50cf3c7-8dce-4079-b0a6-cec6daf25b59] received
celery-general-1        | [2026-06-22 10:21:48,135: INFO/ForkPoolWorker-2] Completed local runtime bootstrap
celery-general-1        | [2026-06-22 10:21:48,136: INFO/ForkPoolWorker-2] Task app.tasks.runtime_bootstrap_tasks.complete_local_runtime_bootstrap[f50cf3c7-8dce-4079-b0a6-cec6daf25b59] succeeded in 0.007374874985544011s: {'status': 'ready', 'primary_market': 'US', 'market': 'US'}
frontend-1              | 192.168.65.1 - - [22/Jun/2026:08:21:49 +0000] "GET /api/v1/runtime/activity HTTP/1.1" 200 583 "http://hic-svnt-macbook.local/operations" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/149.0.0.0 Safari/537.36" "-"
frontend-1              | 192.168.65.1 - - [22/Jun/2026:08:21:53 +0000] "GET /api/v1/app-capabilities HTTP/1.1" 200 4449 "http://hic-svnt-macbook.local/operations" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/149.0.0.0 Safari/537.36" "-"
frontend-1              | 192.168.65.1 - - [22/Jun/2026:08:21:53 +0000] "GET /api/v1/runtime/activity HTTP/1.1" 200 583 "http://hic-svnt-macbook.local/operations" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/149.0.0.0 Safari/537.36" "-"
frontend-1              | 192.168.65.1 - - [22/Jun/2026:08:21:57 +0000] "GET /api/v1/telemetry/alerts?evaluate=true HTTP/1.1" 200 1048 "http://hic-svnt-macbook.local/operations" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/149.0.0.0 Safari/537.36" "-"
frontend-1              | 192.168.65.1 - - [22/Jun/2026:08:21:59 +0000] "GET /api/v1/operations/jobs HTTP/1.1" 200 3572 "http://hic-svnt-macbook.local/operations" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/149.0.0.0 Safari/537.36" "-"

File: backend/app/scanners/scan_orchestrator.py [hunk 1]
```diff
@@ -883,8 +884,20 @@ def _combine_results(
                     avg_volume = int(vol_series.mean())
 
         # Calculate dollar volume if we have both avg_volume and price
-        if avg_volume and current_price:
+        if (
```
Comment (xang1234) [inline backend/app/scanners/scan_orchestrator.py]: This fixes the observed plain `NaN` case. I would consider normalizing both operands once and requiring finite values, then multiplying the normalized values. That also avoids a few nearby scanner-error paths: `inf` passes `math.isnan(...)` and then fails at `int(...)`, `pd.NA` can raise during `float(...)`, and a numeric string can pass the float check but still fail because the original string is multiplied.

```python
def _finite_float(value: object) -> float | None:
    if value is None:
        return None
    try:
        numeric = float(value)
    except (TypeError, ValueError):
        return None
    return numeric if math.isfinite(numeric) else None
```

Then this block can become:

```python
avg_volume_value = _finite_float(avg_volume)
current_price_value = _finite_float(current_price)

if avg_volume_value is not None and current_price_value is not None:
    result["avg_dollar_volume"] = int(avg_volume_value * current_price_value)
else:
    logger.debug(
        "Skipping avg_dollar_volume for %s avg_volume=%s current_price=%s",
        symbol,
        avg_volume,
        current_price,
    )
```

A small regression test around `float("nan")`, `float("inf")`, `pd.NA`, and a numeric string would pin this down nicely.

**Review by xang1234 — COMMENTED** [review]
Thanks. The current change fixes the observed plain-NaN failure path; the proposed tweak below makes the field resilient to  non-finite/coercion cases as well.
